hooks/basic/collector.py

```diff
@@ -538,26 +544,56 @@ def _get_item_type_from_settings(self, settings, path, is_sequence):
 
         # default values used if no specific type can be determined
         item_type = "file.unknown"
+        # work_path_template default value is none so it won't create a context for the item
+        work_path_template = None
 
         # keep track if a common type was identified for the extension
         common_type_found = False
 
-        # look for the extension in the common file type info dict
-        for item_type, type_info in settings["Item Types"].value.iteritems():
+        # mapping to track how many item types contain the given extension
+        template_item_type_mapping = dict()
 
+        # look for the extension in the common file type info dict
+        # using the raw value to get a raw work_path_template
+        for current_item_type, type_info in settings["Item Types"].raw_value.iteritems():
+            tmp_work_path_template = None
             if extension in type_info["extensions"]:
+                # match this raw template against all environments
```

Comment:
I can't really wrap my head around a case where we won't have an extension discriminator, and we use just the `work_path_template` to filter item types, because in that case multiple items will lead to a `matched_work_path_template=None` and the last item_type that matches would win if nothing matches. Since `template_item_type_mapping` dict will store the last item_type.

Also, there is no precedence order that we decided. Do the item type that matches with extensions match, get preference over something that matches without extensions match?

We might need to discuss more what to do in these cases, before implementing this support.

Reply:
Thinking about this more, the description for `extensions` says "A string pattern to match a file extension." which suggests to me that we should be using a matching function rather than an equality to evaluate it. I'd like to update this instead to the following:

```python
ext_matched = False
for ext_filter in type_info.get("extensions", []):
    if fnmatch.fnmatch(extension, ext_filter):
        ext_matched = True
        break

if not ext_matched:
    continue
```

This would then allow a user to configure an open match such as `['*']` if desired.

Reply:
That sounds much better than making it an optional field. Will implement the pattern matching.